gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/impl/sql/execute/CreateSchemaConstantAction.java

```diff
@@ -177,9 +178,13 @@ private void executeConstantActionMinion(Activation activation,
 
 		//if the schema descriptor is an in-memory schema, we donot throw schema already exists exception for it.
 		//This is to handle in-memory SESSION schema for temp tables
+		// Special case INTERNAL schema
 		if ((sd != null) && (sd.getUUID() != null))
 		{
-			throw StandardException.newException(SQLState.LANG_OBJECT_ALREADY_EXISTS, "Schema" , schemaName);
+			if (!schemaName.equalsIgnoreCase(StoreCallbacks.INTERNAL_SCHEMA_NAME))
```

Comment:
Why exactly is this special check required? Would it be better to check and create at higher-level in Snappy layer when creating a column table? Normally a schema gets created implicitly so I think snappy layer should not even need to explicitly create the schema i.e. a DDL like "CREATE TABLE SYSCS_INTERNAL.C1 ... " will create the SYSCS_INTERNAL schema automatically on first execution.

The only reason to create it explicitly would be for permissions to the schema if authorization is turned on for GemXD (not a concern for preview release). For that case it will be better to implicitly create the schema in GemXD boot itself like other internal schemas (like in DataDictionaryImpl.addSystemSchema) and provide PUBLIC permissions to the schema. However, PUBLIC permissions to a schema is not possible right now without special workaround, so we need to discuss that later.